app/controllers/course/material/controller.rb

```diff
@@ -0,0 +1,33 @@
+class Course::Material::Controller < Course::ComponentController
+  load_and_authorize_resource :folder, through: :course, through_association: :material_folders,
+                                       class: Course::Material::Folder.name
+
+  before_action :check_component
+  before_action :add_folder_breadcrumb
+
+  private
+
+  # @return [Course::MaterialsComponent|nil] The materials component or nil if disabled.
+  def component
+    current_component_host[:course_materials_component]
+  end
+  helper_method :component
```

Comment:
@lowjoel How about it now ? I have made this a helper_method, and we can get settings through `component.settings`